### PR TITLE
RPC:golint fix

### PR DESCRIPTION
**Pull request by ShadowErii**
However,i don't get why if i return error last in a method,it will increse the nums the reciever.Types().Method()(including that method),which will cause getting more callbacks

### CODE REVIEW

**Comment by ucwong** [comment]
Which line ？

File: rpc/testservice_test.go [hunk 1]
```diff
@@ -93,8 +93,8 @@ func (s *testService) Rets() (string, error) {
 }
 
 //lint:ignore ST1008 returns error first on purpose.
-func (s *testService) InvalidRets1() (string, error) {
-	return "", nil
+func (s *testService) InvalidRets1() (error, string) {
```
Comment (ShadowErii) [inline rpc/testservice_test.go]: ignore the ST1008 check here , and then it won't include in the callbacks

Comment (ucwong) [reply rpc/testservice_test.go]: InvalidRets1 is made as wrong format method for testing

Comment (ucwong) [reply rpc/testservice_test.go]: So we expect it is wrong

File: rpc/http.go [hunk 2]
```diff
@@ -30,6 +30,8 @@ import (
 	"time"
 )
 
+type contextKey string
+
```
Comment (ucwong) [inline rpc/http.go:34]: Is it necessary to use a cname here

Comment (ShadowErii) [reply rpc/http.go:34]: yes,it's better not to use string as key in the context.withvalue() ,i thought

Comment (ucwong) [reply rpc/http.go:34]: OK, it seems from golint rules

Comment (ucwong) [reply rpc/http.go:34]: It will be better to use name of  ```httpContextKey or rpcContextKey``` with more meanings than ```contextKey``` here .

Comment (ShadowErii) [reply rpc/http.go:34]: ok,
it avoid key collisions by defining a separate type,although may never happen :)

**Pull request merged**